internal/dashboard/root.go

```diff
@@ -0,0 +1,227 @@
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(cosmov1alpha1.AddToScheme(scheme))
+	// +kubebuilder:scaffold:scheme
+}
+
+type options struct {
+	KubeConfigPath string
+	KubeContext    string
+
+	ZapOpts zap.Options
+
+	Logr *clog.Logger
+	// Client *kosmo.Client
+
+	StaticFileDir           string
+	CookieDomain            string
+	CookieHashKey           string
+	CookieBlockKey          string
+	CookieSessionName       string
+	ResponseTimeoutSeconds  int64
+	GracefulShutdownSeconds int64
+	TLSPrivateKeyPath       string
+	TLSCertPath             string
+	Insecure                bool
+	ServerPort              int
+	MaxAgeMinutes           int
+}
+
+func NewRootCmd(o *options) *cobra.Command {
+
+	rootCmd := &cobra.Command{
+		Use:   "authserver",
+		Short: "Authenticate server",
+		Long: `
+cosmo dashboard server
+Complete documentation is available at http://github.com/cosmo-workspace/cosmo
+
+MIT 2023 cosmo-workspace/cosmo
+`,
+		Version:           "v0.8.0 cosmo-workspace 2023",
+		PersistentPreRunE: o.PreRunE,
+		RunE:              o.RunE,
+	}
+
+	goflags := flag.NewFlagSet(os.Args[0], flag.ExitOnError)
+	klog.InitFlags(goflags)
+	o.ZapOpts.BindFlags(goflags)
+	ctrl.RegisterFlags(goflags)
+	rootCmd.Flags().AddGoFlagSet(goflags)
+
+	rootCmd.PersistentFlags().Int64Var(&o.ResponseTimeoutSeconds, "timeout-seconds", 3, "Timeout seconds for response")
+	rootCmd.PersistentFlags().Int64Var(&o.GracefulShutdownSeconds, "graceful-shutdown-seconds", 10, "Graceful shutdown seconds")
+	rootCmd.PersistentFlags().StringVar(&o.StaticFileDir, "serve-dir", "/app/public", "Static file dir to serve")
+	rootCmd.PersistentFlags().StringVar(&o.CookieDomain, "cookie-domain", "", "Cookie domain name")
+	rootCmd.PersistentFlags().StringVar(&o.CookieHashKey, "cookie-hashkey", "", "Cookie hashkey")
+	rootCmd.PersistentFlags().StringVar(&o.CookieBlockKey, "cookie-blockkey", "", "Cookie blockkey")
+	rootCmd.PersistentFlags().StringVar(&o.CookieSessionName, "cookie-session-name", "cosmo-auth", "Cookie session name")
+	rootCmd.PersistentFlags().StringVar(&o.TLSPrivateKeyPath, "tls-key", "tls.key", "TLS key file path")
+	rootCmd.PersistentFlags().StringVar(&o.TLSCertPath, "tls-cert", "tls.crt", "TLS certificate file path")
+	rootCmd.PersistentFlags().BoolVar(&o.Insecure, "insecure", false, "start http server not https server")
+	rootCmd.PersistentFlags().IntVar(&o.ServerPort, "port", 8443, "Port for dashboard server")
+	rootCmd.PersistentFlags().IntVar(&o.MaxAgeMinutes, "maxage-minutes", 720, "session maxage minutes")
+
+	return rootCmd
+}
+
+func (o *options) PreRunE(cmd *cobra.Command, args []string) error {
+	if err := o.Validate(cmd, args); err != nil {
+		return fmt.Errorf("validation error: %w", err)
+	}
+	if err := o.Complete(cmd, args); err != nil {
+		return fmt.Errorf("invalid options: %w", err)
+	}
+	return nil
+}
+
+func (o *options) Validate(cmd *cobra.Command, args []string) error {
+	if o.CookieDomain == "" {
```

Comment:
```suggestion
	// TODO: validate length
	if o.CookieDomain == "" {
```

Reply:
If cookie-domain is not specified, cookies can only be enabled between the same hosts.
This is assuming that the host is localhost, etc.